homeassistant/components/vesync/sensor.py

```diff
@@ -121,6 +122,47 @@ def ha_dev_type(device):
         update_fn=update_energy,
         exists_fn=lambda device: ha_dev_type(device) == "outlet",
     ),
+    VeSyncSensorEntityDescription(
+        key="energy-weekly",
+        name="energy use weekly",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda device: device.weekly_energy_total,
+        update_fn=update_energy,
+        exists_fn=lambda device: ha_dev_type(device) == "outlet",
+    ),
+    VeSyncSensorEntityDescription(
+        key="energy-monthly",
+        name="energy use monthly",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda device: device.monthly_energy_total,
+        update_fn=update_energy,
+        exists_fn=lambda device: ha_dev_type(device) == "outlet",
+    ),
+    VeSyncSensorEntityDescription(
+        key="energy-yearly",
+        name="energy use yearly",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda device: device.yearly_energy_total,
+        update_fn=update_energy,
+        exists_fn=lambda device: ha_dev_type(device) == "outlet",
+    ),
+    VeSyncSensorEntityDescription(
+        key="voltage",
+        name="current voltage",
+        device_class=SensorDeviceClass.VOLTAGE,
+        native_unit_of_measurement=ELECTRIC_POTENTIAL_VOLT,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
```

Comment:
Normally we don't categorize voltage sensors as diagnostic. What does this measurement represent?

Reply:
Voltage is the AC line voltage at the outlet. I have no need to make this diagnostic, so I'll just remove that setting.

Reply:
I updated the code to remove diagnostic category for Voltage, so I'll resolve this conversation and re-ask for review.
Thanks!